flax/linen/recurrent.py

```diff
@@ -651,16 +664,27 @@ def __call__(
         else it will return a tuple of the final carry and the output sequence.
       time_major: if ``time_major=False`` (default) it will expect inputs with shape
         ``(*batch, time, *features)``, else it will expect inputs with shape ``(time, *batch, *features)``.
+      reverse: overrides the ``reverse`` attribute, if ``reverse=False`` (default) the sequence
+        is processed from left to right, else it will be processed from right to left, output order
+        will be the same as input order. If ``segmentation_mask`` is passed, padding elements will
```

Comment:
I think `keep_order` is still a bit difficult to understand, so good documentation is important. This might be too unclear:

```
reverse: overrides the ``reverse`` attribute, if ``reverse=False`` (default) the sequence
        is processed from left to right, else it will be processed from right to left, output order
        will be the same as input order.
```
what about:

reverse: overrides the ``reverse`` attribute, if ``reverse=False`` (default) the sequence
        is processed from left to right and returned in the original order, else it will be processed from right to left, and returned in reverse order (but with the padding values at the end).
```

maybe even a simple example could be good. e.g. 
`[1, 2, 3, 0, 0]` vs `[3, 2, 1, 0, 0]`?